server.js

```diff
@@ -28,31 +28,92 @@ app.get('/posts/:pid', (req, res) => {
         let matchingPost = posts.find( ({pid}) => pid == requestedPostId);
         if(!matchingPost) { throw new Error(`Sorry we don't have a post id of ${requestedPostId}`)}
         res.json(matchingPost);
-        res.send(console.log(`You can now see the post ${pid}`))
+        //res.send(`You can now see the post ${pid}`)
     } catch (error) {
         res.status(404).json({message: error.message});
     }
     
 });
 
-// CREATE - one post -need to add item persistently
+// CREATE - one post -will need to add item persistently
+// CLIENT INPUT LOOKS LIKE:
+// { 
+//     "title": "new post, new title",
+//     "message": "new post, new message",
+//     "giphy": "https://media0.giphy.com/media/enNk8yXXpDHYzD3tqF/giphy.gif?cid=5c2583098j8m18kl0gh9xq3wv558ovtq7d8ykr71wna9ycwg&rid=giphy.gif&ct=g",
+// }
+
 app.post('/posts', (req, res) =>{
-    console.log("hello");
     try{
-    let newPost = req.body;
-    console.log(posts);
-    if(!newPost) { throw new Error(`No comment entered`)}
+    let newId = posts[0].pid + 1;
+    let newTitle = req.body.title;
+    let newMessage = req.body.message;
+    let newGiphy = req.body.giphy;
+    let timeNow = Date.now();
+
+    let newPost = {
+        "pid": newId,
+        "title": newTitle,
+        "message": newMessage,
+        "giphy": newGiphy,
+        "comments": [],
+        "reactions": {
+            "thumbs_up": [],
+            "thumbs_down": [],
+            "heart": []
+        },
+        "time": timeNow
+    }
+    if(!newTitle) { throw new Error(`No title entered.`)};
+    if(!newTitle) { throw new Error(`No text entered.`)};
     posts.unshift(newPost);
-    console.log(posts);
-    res.json(posts[0]);
+    res.json(newPost); // only one res. will go through
+    //res.send("comment added successfully"); // only one res. will go through
     } catch (error) {
         res.status(400).json({message: error.message});
     }
 });
 
-// CREATE - one comment / UPDATE the post with one comment
-app.post('/posts', (req, res) => {
+// CREATE - add one comment 
+
+// CLIENT INPUT LOOKS LIKE:
+// { 
+//     "cid": 2,
+//     "comment": "another comment"
+// }
 
+// WILL NEED TO FILL IN AN ELEMEMENT that look like:
+// "comments": [
+//     {
+//         "cid": 3,
+//         "message": "message",
+//         "time": 1285253453498
+//     }
+// ]
+
+app.post('/posts/:id/comments', (req, res) => {
+    try {
+        let newMessage = req.body.message;
+        console.log(posts);
+        console.log(newMessage);
+        let timeNow = Date.now();
+        let requestedPostId = req.body.pid;
```

Comment:
```suggestion
        let requestedPostId = req.params.id;
```
id from the url path

Reply:
Can ```req``` take arguments both from the url path and the body at the same time? from a quick trial, it seems not so

Reply:
`req.body` is your request body, `req.params` are your url parameters, `req` has both of these properties

Reply:
addressed